WIP Freestar Ads

GregJunkaTR opened the pull request:
### What's in this pull request

- [ ] Bug fix
- [x] New component/feature
- [ ] Documentation update
- [ ] Other

### Description

Adds Freestar Ads

### Before submitting, please check that you've

- [x] Read our [contributing guide](https://github.com/reuters-graphics/graphics-components/blob/master/CONTRIBUTING.md) at some point
- [x] Formatted you code correctly (i.e., prettier cleaned it up)
- [x] Documented any new components or features
- [x] Tagged an editor to review

<!-- GITHUB_RELEASE PR BODY: canary-version -->

The ad flow should be following based on my understanding:

1. Load bootstrap script
2. When the above is loaded do freestar object setup and register initCallback of bootstrap
3. When bootstrap work finishes, load pubfig.min.js
4. Then it will load other ad related scripts and turn on ads on the site

hobbes7878 commented:
@GregJunkaTR Couple Qs:

The story for the AdSlot component doesn't actually display any ads when running from the whitelisted localhost domain, localhost.arcpublishing.com. Is that expected behaviour? Where will this run as expected?

Can you please confirm the explicit placementName types I've added: https://github.com/reuters-graphics/graphics-components/blob/greg-ads/src/components/AdSlot/AdSlot.svelte#L4-L7

Is the only valid dataFreestarAd `__970x250` or is that variable in the future?

GregJunkaTR commented:
> @GregJunkaTR Couple Qs:
> 
> The story for the AdSlot component doesn't actually display any ads when running from the whitelisted localhost domain, localhost.arcpublishing.com. Is that expected behaviour? Where will this run as expected?
> 
> Can you please confirm the explicit placementName types I've added: https://github.com/reuters-graphics/graphics-components/blob/greg-ads/src/components/AdSlot/AdSlot.svelte#L4-L7
> 
> Is the only valid dataFreestarAd `__970x250` or is that variable in the future?

Yes, no ads are expected. They will only display if they are configured properly on the Ads Server, and also they never work for me when testing locally. That's why we need you to set up a test page using a public domain. But even locally you can inspect the places where the ads would be shown to see if the markup is being replaced. I was showing you during our meeting that the markup is being replaced, which means that the code works, just ads are not served.

The variables/placement types aren't important at this point, they will need to be configured on the ad server and updated in the code once the page is deployed to a public domain. I don't know how it works exactly but I think they are configured per domain, thus they wouldn't work anyway until the page is deployed.

hobbes7878 commented:
Preview of page with ads published here: 
https://www.reuters.com/graphics/HEALTH-CORONAVIRUS/TEST/byvrklraape/

(Please mind that public link doesn't stray too far...)

No ads displaying, but Ima assume this because there's some config needed on the ads server side.
